pkg/config/env.go

```diff
@@ -0,0 +1,20 @@
+package config
+
+import (
+	"os"
+
+	"k8s.io/klog/v2"
+)
+
+const (
+	PodNamespaceEnvKey = "POD_NAMESPACE"
+)
+
+// GetPodNamespace gets the namespace of the pod.
+func GetPodNamespace() string {
+	podNamespace := os.Getenv(PodNamespaceEnvKey)
+	if podNamespace == "" {
+		klog.Errorf("environment variable %v not found", PodNamespaceEnvKey)
```

Comment:
Do not add new logger package here. We don't use klog.

Reply:
What log should we add.

Reply:
Its actually a FATAL error in our case. If this does not exist, then we should continue further.

The log message you added does not crash the pod. Suppose namespace is not set, what would happen to the watcher? will it throw an error? If it throws an error, then we will be in state where we are continuously calling resetting the watcher object.

Reply:
In our class we handle this error.

Reply:
Solved.